lib/core/test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -434,6 +436,34 @@ expectPathEventuallyExist filepath = do
             False ->
                 threadDelay (oneSecond `div` 2) >> doesPathExistNow
 
+-- | A file is eventually created on the given location
+expectProcStdOutHas :: CreateProcess -> Text -> IO ()
+expectProcStdOutHas procc out = do
+    handle <- async (runProcUntil procc out)
+    winner <- race (threadDelay (60 * oneSecond)) (wait handle)
+    case winner of
+        Left _ -> expectationFailure $
+            "waited more than 60s for proccess to have in stdout = " ++ T.unpack out
+        Right _ ->
+            return ()
+  where
+    oneSecond = 1_000_000
+
+    runProcUntil :: CreateProcess -> Text -> IO ()
+    runProcUntil pr wants = withCreateProcess pr $ \_ (Just stdout) _ h -> do
```

Comment:
One minor thing but there a partial pattern-match here `Just stdout` which will fail hard with a non-helpful error if it fails. We only have a `Just` handle if the process was created with `CreatePipe` option, though, here, since we receive the `CreateProcess` as an argument, there's no way to _know_ that. I bet the call-site did it right, but, it'll be safer to either, overwrite the stdout stream parameter to be `CreatePipe` before calling `withCreateProcess`, or, to leave the creation of the process to the expect function.

Reply:
Ok, went for: `overwrite the stdout stream parameter to be CreatePipe`